File: leetcode/practice/728-self-dividing-numbers/naive.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        def selfDividing(ele) :
            for ch in str(ele) :
                if ch == '0' :
                    return False ;
                
                if ele%int(ch) != 0 :
                    return False 
                
            return True
        ans = []
        for ele in range(left , right + 1) :
            if ele < 10 :
                ans.append(ele)
            elif selfDividing(ele) :
                ans.append(ele)
            
        return ans

File: leetcode/practice/728-self-dividing-numbers/naive.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left , int right) ;
    bool selfDividing(int num) ;
};

bool Solution :: selfDividing(int num)
{
    string n = to_string(num) ;
    for(auto ch : n)
    {
        if(ch == '0')
            return false ;
        if(num%(ch - '0') != 0)
            return false ;
    }
    return true ;
}

vector<int> Solution :: selfDividingNumbers(int left , int right)
{
    vector<int>ans ;
    for(int num = left ; num <= right ; num++)
    {
        if(num < 10)
            ans.push_back(num) ;
        else if(selfDividing(num))
            ans.push_back(num) ;

    }
    return ans ;
}
